### PR TITLE
Feature/update realm

### LINKED ISSUE #1926: Upgrade realm to 2.X.X version
We are running `^.0.14.3` and last version of realm is `1.2` something. `1.0.0` was released early this year. We need to upgrade to avoid semantic issues in RealmJS errors before we can build and run the app on Xcode.

See https://github.com/status-im/status-react/issues/1923

Type: Bug

### PR DESCRIPTION
**Pull request by yenda**
fixes #1926

## Summary

extending the type of js objects doesn't work anymore so we convert it to json then import it to cljs with transit. Sounds crazy but it is actually faster than doing a js->clj call.

## Steps to test

This touches everything related to the database: contacts, messages, discover, network

status: ready

### CODE REVIEW

File: src/status_im/data_store/realm/core.cljs [hunk 1]
```diff
@@ -121,81 +123,46 @@
...
+(def reader (transit/reader :json))
+
+(defn js-object->clj
+  "Converts any js type/object into a map recursively
+  Performs 5 times better than iterating over the object keys
+  and that would require special care for collections"
+  [js-object]
+  (let [o (->> js-object
+               (.stringify js/JSON)
+               (transit/read reader))]
```
Comment (jeluard) [inline src/status_im/data_store/realm/core.cljs:138]: Are we sure this always work as intended? `transit/read` is supposed to accept the result of `transit/write` as argument and not just regular JSON.

**Review by jeluard — COMMENTED** [review]
I'd love to see some more tests here :)

File: src/status_im/data_store/realm/core.cljs (hunk 1 above)
Comment (janherich) [reply src/status_im/data_store/realm/core.cljs:138]: This is quite strange, are we sure that library like `cheshire` or `clj-json` is not both faster and better suited to the task of just deserialising json character stream into clojure datastructures ?
I love transit, but this doesn't seems like the best use case for it.

Comment (yenda) [reply src/status_im/data_store/realm/core.cljs:138]: Just deserialising json characters stream into Clojure datastructure is exactly what transit reader is doing

**Comment by yenda** [comment]
@jeluard there will be tests in the refactoring for now the idea was just to do the minimal thing to make an update possible, so that XCode update for instance is unblocked. I have yet to see if and how we can test realm outside of the mobile though

File: src/status_im/data_store/realm/core.cljs (hunk 1 above)
Comment (yenda) [reply src/status_im/data_store/realm/core.cljs:138]: https://cljs.github.io/api/cljs.core/js-GTclj

`Note that js->clj is not optimized for speed and the transit.cljs library is recommended for parsing large amounts of JSON data.`

**Comment by jeluard** [comment]
@yenda Transit uses JSOn as a vehicle but with a specific semantic (see [transit format](https://github.com/cognitect/transit-format)). There is no garanty that JSON -> Clojure conversion will fit our use case.

There is no rush to unblock other migration. Please add tests to make sure we don't introduce regression.

**Review by jeluard — CHANGES_REQUESTED** [review]
Expecting tests for low-level infrastructure changes.

**Review by janherich — APPROVED** [review]

File: src/status_im/data_store/realm/core.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/data_store/realm/core.cljs:138]: Sounds good then.

**Review by jeluard — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Pull request merged**